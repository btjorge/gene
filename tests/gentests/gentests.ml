let prefix = "../progs/"

let progs = [ "max02"; "max03"; "max04"; "max05"; "max06"; "max3"; "max_imp";
              "max_imp2" ]

let symbinterp file =
  let command = "../../bin/gene.exe --generate-tests " ^ prefix ^ file in
  print_endline command;
  if Sys.command command <> 0
  then failwith "error";
  ()

let () =
  List.iter symbinterp progs
