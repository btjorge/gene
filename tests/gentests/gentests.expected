../../bin/gene.exe --generate-tests ../progs/max02
The following test parameters are not used by the code (see the generic test):
_x0, _x1
Generalized test:
assertEquals (max (_x0, _x1), _x2)
Your test:
assertEquals (max (5, 2), 5)
Passing tests:
assertEquals (max (_x0, _x1), 5)
Failing tests:
assertEquals (max (_x0, _x1), 0)
../../bin/gene.exe --generate-tests ../progs/max03
The following test parameters are not used by the code (see the generic test):
_x1
Generalized test:
assertEquals (max (_x0, _x1), _x2)
Your test:
assertEquals (max (5, 2), 5)
Passing tests:
assertEquals (max (0, _x1), 0)
Failing tests:
assertEquals (max (0, _x1), 1)
../../bin/gene.exe --generate-tests ../progs/max04
The following test parameters are not used by the code (see the generic test):
_x0
Generalized test:
assertEquals (max (_x0, _x1), _x2)
Your test:
assertEquals (max (5, 2), 5)
Passing tests:
assertEquals (max (_x0, 0), 0)
Failing tests:
assertEquals (max (_x0, 0), 1)
../../bin/gene.exe --generate-tests ../progs/max05
Your test:
assertEquals (max (5, 2), 5)
Passing tests:
assertEquals (max (0, 0), 0)
assertEquals (max (1, 0), 0)
Failing tests:
assertEquals (max (0, 0), 2)
assertEquals (max (0, -1), 3)
../../bin/gene.exe --generate-tests ../progs/max06
Your test:
assertEquals (max (5, 2), 5)
Passing tests:
assertEquals (max (0, 0), 0)
assertEquals (max (0, -1), 0)
Failing tests:
assertEquals (max (0, 0), 2)
assertEquals (max (1, 0), 3)
../../bin/gene.exe --generate-tests ../progs/max3
Your test:
assertEquals (max3 (1, 3, 2), 3)
Passing tests:
assertEquals (max3 (0, 0, 0), 0)
assertEquals (max3 (0, 0, -1), 0)
assertEquals (max3 (0, -1, 0), 0)
assertEquals (max3 (0, -1, -1), 0)
Failing tests:
assertEquals (max3 (0, 0, 0), 3)
assertEquals (max3 (0, 0, -1), 3)
assertEquals (max3 (0, -1, 0), 2)
assertEquals (max3 (0, -1, -1), 2)
../../bin/gene.exe --generate-tests ../progs/max_imp
Your test:
test () {
  x := 2;
  y := 3;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 3)
}
Passing tests:
test (_x0, _x1, _x2) {
  x := 0;
  y := 0;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 0)
}
test (_x0, _x1, _x2) {
  x := 0;
  y := -1;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 0)
}
Failing tests:
test (_x0, _x1, _x2) {
  x := 0;
  y := 0;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 2)
}
test (_x0, _x1, _x2) {
  x := 1;
  y := 0;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 3)
}
../../bin/gene.exe --generate-tests ../progs/max_imp2
Your test:
test () {
  x := 2;
  y := 3;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, y)
}
Passing tests:
test (_x0, _x1) {
  x := 0;
  y := 0;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, y)
}
Failing tests:
test (_x0, _x1) {
  x := 1;
  y := 0;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, y)
}
